### DISCUSSION TITLE
[HELP] How to fix the severity vulnerabilities from  size-limit

### DISCUSSION THREAD

**Discussion by ocylib** [jaredpalmer/tsdx#1063] [Q&A]
I am new to tsdx, when run `npm install`, the log showed me there were ` 7 moderate severity vulnerabilities`

> PS D:\dev\tszl> npm install
> tszl@0.1.0 prepare
> tsdx build        
> @rollup/plugin-replace: 'preventAssignment' currently defaults to false. It is recommended to set this option to `true`, as the 
> next major version will default this option to `true`.
> 
> @rollup/plugin-replace: 'preventAssignment' currently defaults to false. It is recommended to set this option to `true`, as the next major version will default this option to `true`.
> ✓ Creating entry file 4.8 secs
> ✓ Building modules 9.5 secs
> 
> up to date, audited 1408 packages in 26s
> 
> 104 packages are looking for funding
>   run `npm fund` for details        
> 
> 7 moderate severity vulnerabilities
> 
> To address all issues, run:        
>   npm audit fix
> 
> Run `npm audit` for details.

Then I ran `npm audit` and got logs as follow. `IT SEEEMS SOME PACKAGES USED BY TSDX CAUSED THIS`

> PS D:\dev\tszl> npm audit
> # npm audit report
> 
> glob-parent  <5.1.2
> Severity: moderate
> Regular expression denial of service - https://npmjs.com/advisories/1751
> fix available via `npm audit fix`
> node_modules/watchpack-chokidar2/node_modules/glob-parent
>   chokidar  1.0.0-rc1 - 2.1.8
>   Depends on vulnerable versions of glob-parent
>   node_modules/watchpack-chokidar2/node_modules/chokidar
>     watchpack-chokidar2  *
>     Depends on vulnerable versions of chokidar
>     node_modules/watchpack-chokidar2
>       watchpack  1.7.2 - 1.7.5
>       Depends on vulnerable versions of watchpack-chokidar2
>       node_modules/watchpack
>         webpack  4.44.0 - 4.46.0
>         Depends on vulnerable versions of watchpack
>         node_modules/webpack
>           @size-limit/webpack  >=4.5.6
>           Depends on vulnerable versions of webpack
>           node_modules/@size-limit/preset-small-lib/node_modules/@size-limit/webpack
>             @size-limit/preset-small-lib  >=4.5.6
>             Depends on vulnerable versions of @size-limit/webpack
>             node_modules/@size-limit/preset-small-lib
> 
> 7 moderate severity vulnerabilities
> 
> To address all issues, run:
>   npm audit fix

And then, I ran `npm audit fix`, the log showed errors as follow. 

> PS D:\dev\tszl> npm audit fix
> npm ERR! code ERESOLVE
> npm ERR! ERESOLVE unable to resolve dependency tree
> npm ERR!
> npm ERR! While resolving: tszl@0.1.0
> npm ERR! Found: size-limit@5.0.1
> npm ERR! node_modules/size-limit
> npm ERR!   dev size-limit@"^5.0.1" from the root project
> npm ERR! 
> npm ERR! Could not resolve dependency:
> npm ERR! peer size-limit@"4.5.5" from @size-limit/preset-small-lib@4.5.5
> npm ERR! node_modules/@size-limit/preset-small-lib
> npm ERR!   dev @size-limit/preset-small-lib@"^4.5.5" from the root project
> npm ERR!
> npm ERR! Fix the upstream dependency conflict, or retry
> npm ERR! this command with --force, or --legacy-peer-deps
> npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
> npm ERR!
> npm ERR! See C:\Users\ziche\AppData\Local\npm-cache\eresolve-report.txt for a full report.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     C:\Users\ziche\AppData\Local\npm-cache\_logs\2021-07-16T01_23_12_966Z-debug.log

I have tried many times, nothing change.
What can I do to fix this, thanks for any help, please.

**Comment by aladdin-add** [comment]
this is not something related to tsdx - it's not depended on size-limit. you can run `npm why size-limit` to see why it's installed.
